lightning/src/ln/channelmanager.rs

```diff
@@ -16855,6 +16819,38 @@ where
 	}
 }
 
+// If the HTLC corresponding to `prev_hop_data` is present in `decode_update_add_htlcs`, remove it
+// from the map as it is already being stored and processed elsewhere.
+fn dedup_decode_update_add_htlcs<L: Deref>(
+	decode_update_add_htlcs: &mut HashMap<u64, Vec<msgs::UpdateAddHTLC>>,
+	prev_hop_data: &HTLCPreviousHopData, removal_reason: &'static str, logger: &L,
+) where
+	L::Target: Logger,
+{
+	decode_update_add_htlcs.retain(|src_outb_alias, update_add_htlcs| {
```

Comment:
I know this was a move-only, but this should be optimized by just using `entry` on the `HashMap`.